raft-tech/TANF-app
# 2700 - Migration discrepancy problem

## Summary of Changes
_Provide a brief summary of changes_
Pull request closes #2700 _

This PR resolves the migration discrepancy that happened in Sprint 80.

## How to Test

The test can be done against two states:

1. Fresh deployment: this can be tested against local dev environment. The app should run all migrations and the app shall run successfully
2. Deployment against prod and staging. (Note: Staging had been checked and DB state was matching what we have in prod : Please see [this](https://hackmd.io/Iqj52CuCRUicwQ1B8B4MJQ) for more information). As can be seen below, the PR was deployed successfully with no error message and the app is running.

<img width="1209" alt="Screenshot 2023-09-15 at 7 08 12 AM" src="https://github.com/raft-tech/TANF-app/assets/97037188/d671b7cf-4266-45cb-a4a6-ce6358cd00e4">

The image below shows results from two commands. 

The section on the top shows result from `./manage.py showmigrations search_indexes`. This command compares the migration in DB against what is existing in local ~/app directory.

The lower part of screen shows the result of query from DB in shell.

<img width="1437" alt="Screenshot 2023-09-15 at 7 00 13 AM" src="https://github.com/raft-tech/TANF-app/assets/97037188/e9f8c041-29e0-433f-a0cc-a9304fefada8">



## Deliverables

Checklist of ACs:
+ [x] App can runs successfully both locally and when deployed to Cloud.gov
+ [x] App can be deployed to Staging instance without manipulating the migrations manually

### [Deliverable 2: Tested Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
+ Are code coverage minimums met?

### [Deliverable 3: Properly Styled Code](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)


### [Deliverable 4: Accessible](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)



### [Deliverable 5: Deployed](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)


### [Deliverable 6: Documented](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)


### [Deliverable 7: Secure](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)


### [Deliverable 8: User Research](https://github.com/raft-tech/TANF-app/blob/develop/docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
